pandora-client-web/src/components/wardrobe/modules/wardrobeModuleLockSlot.tsx

```diff
@@ -468,3 +512,58 @@ function PasswordInput({
 		</>
 	);
 }
+
+function TimerInput({
+	onChange,
+	timer,
+	pendingAttempt = false,
+}: {
+	onChange: (newValue: number) => void;
+	timer: Immutable<NonNullable<LockSetup['timer']>>;
+	pendingAttempt?: boolean;
+	showInvalidWarning?: boolean;
+}) {
+	const id = useId();
+
+	const [hours, setHours] = useState<number>(0);
+	const [minutes, setMinutes] = useState<number>(0);
+
+	const maxHours = useMemo(() => {
+		return timer.maxMinutes / 60;
+	}, [timer]);
+
+	const maxMinutes = useMemo(() => {
+		return timer.maxMinutes < 59 ? timer.maxMinutes : 59;
+	}, [timer]);
+
+	useEffect(() => {
+		const minutesTotal = hours * 60 + minutes;
+		onChange(minutesTotal > timer.maxMinutes ? timer.maxMinutes : minutesTotal);
+	}, [timer, onChange, hours, minutes]);
+
+	return (
+		<Row className='WardrobeInputRow'>
+			<label htmlFor={ id }>
+				Timer
+			</label>
+			<NumberInput
+				min={ 0 }
+				max={ maxHours }
+				step={ 1 }
+				value={ hours }
+				onChange={ setHours }
+				disabled={ pendingAttempt }
+			/>
+			<label>Hours</label>
+			<NumberInput
+				min={ 0 }
+				max={ maxMinutes }
+				step={ 1 }
+				value={ minutes }
+				onChange={ setMinutes }
+				disabled={ pendingAttempt }
+			/>
+			<label>Minutes</label>
+		</Row>
```

Comment:
The layout of this is a bit weird. I would recommend doing the following instead:
```tsx
return (
	<Row className='WardrobeInputRow' alignY='center'>
		<label>
			Timer
		</label>
		<Row alignY='center'>
			<NumberInput
				id={ `${id}-hours` }
				...
			/>
			<label htmlFor={ `${id}-hours` }>Hours</label>
			{ ' : ' }
			<NumberInput
				id={ `${id}-minutes` }
				...
			/>
			<label htmlFor={ `${id}-minutes` }>Minutes</label>
			{ ' : ' }
			<NumberInput
				id={ `${id}-seconds` }
				...
			/>
			<label htmlFor={ `${id}-seconds` }>Seconds</label>
		</Row>
	</Row>
);
```